py/desispec/scripts/tsnr_afterburner.py

```diff
@@ -0,0 +1,197 @@
+import os
+import glob
+import itertools
+import argparse
+import astropy.io.fits as fits
+import numpy as np
+
+from   desispec.io import read_sky
+from   desispec.io import read_fiberflat
+from   pathlib import Path
+from   desispec.io.meta import findfile, specprod_root
+from   desispec.calibfinder import CalibFinder
+from   desispec.io import read_frame
+from   desispec.io import read_fibermap
+from   desispec.io.fluxcalibration import read_flux_calibration
+from   desiutil.log import get_logger
+from   desispec.tsnr import calc_tsnr
+from   astropy.table import Table, vstack
+
+def parse(options=None):
+    parser = argparse.ArgumentParser(description="Apply fiberflat, sky subtraction and calibration.")
+    parser.add_argument('--outdir', type = str, default = None, required=True,
+                        help = 'Dir. to write to.')
+    parser.add_argument('--prod', type = str, default = None, required=True,
+                        help = 'Path to reduction, e.g. /global/cfs/cdirs/desi/spectro/redux/blanc/')
+    parser.add_argument('--camera', type = str, default = None, required=False,
+                        help = 'Camera to reduce.')
+    parser.add_argument('--summary_only', type = int, default = 0, required=False,
+                        help = 'Write only summary file.')
+    parser.add_argument('--expids', type = str, default = None, required=False,
+                        help = 'Comma separated list of exp ids to be reduced.')
+    args = None
+    if options is None:
+        args = parser.parse_args()
+    else:
+        args = parser.parse_args(options)
+    return args
+
+args=parse()
```

Comment:
All of the code from here on down in this file should be wrapped in a function so that `import desispec.scripts.tsnr_afterburner` doesn't trigger it trying to run a bunch of code.  That could happen e.g. by a test runner looking for tests, or a documentation scraper looking for every function to auto-generate docs.  It's ok to have
```
if __name__ == '__main__':
    main()
```
so that this script can be called with `python tsnr_afterburner.py` without putting in the default installed $PATH.